libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1101,6 +1101,44 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 			ArrayUtils(m_context).popStorageArrayElement(*arrayType);
 			break;
 		}
+		case FunctionType::Kind::StringConcat:
+		{
+			_functionCall.expression().accept(*this);
+			vector<Type const*> argumentTypes;
+			vector<Type const*> targetTypes;
+			for (auto const& argument: arguments)
+			{
+				argument->accept(*this);
+				solAssert(argument->annotation().type, "");
+				argumentTypes.emplace_back(argument->annotation().type);
+				if (argument->annotation().type->category() == Type::Category::FixedBytes)
+					targetTypes.emplace_back(argument->annotation().type);
+				else if (
+					auto const* literalType = dynamic_cast<StringLiteralType const*>(argument->annotation().type);
+					literalType && !literalType->value().empty() && literalType->value().size() <= 32
+				)
+					targetTypes.emplace_back(TypeProvider::fixedBytes(static_cast<unsigned>(literalType->value().size())));
+					// looks like it's converting stringLiteral type to FixedBytes. Probably should not do that for string.concat?
```

Comment:
All of this should be removed shouldn't it?

Also I think it would be less code duplication, if you just case distinguished with another ``if (function.kind() == FunctionType::Kind::StringConcat)`` here, but apart from that used a shared
```
case FunctionType::Kind::StringConcat:
case FunctionType::Kind::BytesConcat:
```